### PR TITLE
Fix/#6080 submission email

### LINKED ISSUE #6080: Submission Confirmation Emails and Feedback Reports
During the 1/25/2024 weekly ECMPS Support meeting with EPA, there was a discussion on what to do with Feedback Report Maintenance in CAT.

The PDF feedback report that is included with 1.0 submission confirmation emails is not currently included in 2.0. This could be due to the new CDX process.

The only users that would have access to submission related information from CDX - Submission History would be the submitter themselves.

Need to determine if:

- It's even possible to include something similar to the PDF feedback report attachment in the submission confirmation emails.
- Or, just expand the data currently shown in the emailed Submission Confirmation Report.

### PR DESCRIPTION
**Pull request by yonatan-dp**
## Pull Request
This is mostly complete functionality for sending feedback report emails per ticket 6080.

### CODE REVIEW

File: src/mail/mail-eval.service.ts [hunk 1]
```diff
@@ -161,8 +154,7 @@ export class MailEvalService {
         const newItem: any = {};
         const certEventRecord: QaCertEvent | QaCertEventGlobal | null =
           certRecord.qaCertEventIdentifier &&
-          (await this.returnManager().findOneBy(
-            !isSubmission ? QaCertEvent : QaCertEventGlobal,
+          (await this.returnManager().findOneBy(QaCertEvent,
```
Comment (djw4erg) [inline src/mail/mail-eval.service.ts:157]: I need more information on why this change occurred.

File: src/mail/mail-eval.service.ts [hunk 2]
```diff
@@ -226,8 +215,7 @@ export class MailEvalService {
         const newItem: any = {};
         const teeRecord: QaTee | QaTeeGlobal | null =
           tee.testExtensionExemptionIdentifier &&
-          (await this.returnManager().findOneBy(
-            !isSubmission ? QaTee : QaTeeGlobal,
+          (await this.returnManager().findOneBy(QaTee,
```
Comment (djw4erg) [inline src/mail/mail-eval.service.ts:218]: I need more information on why this change occurred.

File: src/mail/mail-eval.service.ts [hunk 3]
```diff
@@ -295,8 +280,7 @@ export class MailEvalService {
       for (const em of records) {
         const newItem: any = {};
         const emissionsRecord: EmissionEvaluation | EmissionEvaluationGlobal =
-          await this.returnManager().findOneBy(
-            !isSubmission ? EmissionEvaluation : EmissionEvaluationGlobal,
+          await this.returnManager().findOneBy(EmissionEvaluation,
```
Comment (djw4erg) [inline src/mail/mail-eval.service.ts:283]: I need more information on why this change occurred.

File: src/mail/mail-eval.service.ts [hunk 4]
```diff
@@ -620,68 +600,26 @@ export class MailEvalService {
...
     // Build the context for our email --------------------------------------
     templateContext['dateEvaluated'] = this.displayCurrentDate();
     templateContext['cdxUrl'] = this.configService.get<string>('app.cdxUrl');
 
     // Create Monitor Plan Section of Email
-
     const mpRecord: MonitorPlan | MonitorPlanGlobal =
-      await this.returnManager().findOneBy(
-        !isSubmission && !isSubmissionFailure ? MonitorPlan : MonitorPlanGlobal,
+      await this.returnManager().findOneBy( MonitorPlan,
```
Comment (djw4erg) [inline src/mail/mail-eval.service.ts:622]: I need more information on why this change occurred.

File: src/mail/mail-eval.service.ts (hunk 1 above)
Comment (yonatan-dp) [reply src/mail/mail-eval.service.ts:157]: Context:  Prior to this change, there was no functionality to include attachments in submission feedback emails. The existing functionality simply sends an email with minimal content and with no attachment. The new functionality is to change the email content AND to include the attachments. 

Prior to this change, the sendMassEvalEmail method in MailEvalService class handled sending out emails both for evaluations and submissions.  However, the existing functionality (of sending submission feedback emails) was very minimal.  Given the two functionalities are different, I took out the logic of sending submission feedback emails to its own method SubmissionProcessService. sendFeedbackReportEmail.  That change necessitated the modification of MailEvalService.sendMassEvalEmail to remove submission related logic (and retain only evaluation related logic).

The above removed code makes a decision on what schema to use based on whether it is processing submission or evaluation emails.  Since, this method is no longer processing submission feedback emails, I simplified the logic to always use the schema for evaluations (which is the non-global schema "camdecmpswks").

File: src/mail/mail-eval.service.ts (hunk 2 above)
Comment (yonatan-dp) [reply src/mail/mail-eval.service.ts:218]: Same as above...

File: src/mail/mail-eval.service.ts (hunk 3 above)
Comment (yonatan-dp) [reply src/mail/mail-eval.service.ts:283]: Same as above...

File: src/mail/mail-eval.service.ts (hunk 4 above)
Comment (yonatan-dp) [reply src/mail/mail-eval.service.ts:622]: Same as above...

File: src/mail/mail-eval.service.ts (hunk 1 above)
Comment (djw4erg) [reply src/mail/mail-eval.service.ts:157]: Thanks Yonatan!

**Review by djw4erg — APPROVED** [review]

**Review by djw4erg — APPROVED** [review]
